core/web/job_proposals_controller.go

```diff
@@ -0,0 +1,58 @@
+package web
+
+import (
+	"database/sql"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/smartcontractkit/chainlink/core/services/chainlink"
+	"github.com/smartcontractkit/chainlink/core/web/presenters"
+)
+
+// JobProposalsController manages the job proposals
+type JobProposalsController struct {
+	App chainlink.Application
+}
+
+// Reject rejects a job proposal.
+// Example:
+// "POST <application>/job_proposals/<id>/reject"
+func (jpc *JobProposalsController) Reject(c *gin.Context) {
```

Comment:
I think we want to respect the incoming request context here - ideally a user could cancel a slow running request and it would just rollback everything. See https://github.com/smartcontractkit/chainlink/blob/9302e2cdaf30bfb52cdf65d318d9b0c643608f00/core/web/jobs_controller.go#L125 as an example. Same goes for the rpc_handlers actually, I think we want to pass down that context so we can rollback the tx on cancellation